reference/conanfile/attributes.rst

```diff
@@ -905,10 +905,49 @@ This object should be filled in ``package_info()`` method.
 | self.cpp_info.build_modules      | | List of relative paths to build system related utility module files created by the package. Used by   |
 |                                  | | CMake generators to export *.cmake* files with functions for consumers. Defaulted to ``[]`` (empty)   |
 +----------------------------------+---------------------------------------------------------------------------------------------------------+
+| self.cpp_info.components         | **[Experimental]** Dictionary with different components a package may have: libraries, executables...   |
++----------------------------------+---------------------------------------------------------------------------------------------------------+
 
 The paths of the directories in the directory variables indicated above are relative to the
 :ref:`self.package_folder<folders_attributes_reference>` directory.
 
+.. _cpp_info_components_attributes_reference:
+
+Components
+++++++++++
+
+.. warning::
+
+    This is a **experimental** feature subject to breaking changes in future releases.
+
+When you are creating a Conan package, it is recommended to have only one library (*.lib*, *.a*, *.so*, *.dll*...) per package. However,
+especially with third-party projects like Boost, Poco or OpenSSL, they would contain several libraries inside.
+
+Usually those libraries inside the same package depend on each other and modelling the relationship among them is required.
+
+With **components**, you can model libraries and executables inside the same package and how one depends on the other. Each library or
+executable will be one component inside ``cpp_info`` like this (the following case is not a real example):
+
+.. code-block:: python
+
+    def package_info(self):
+        self.cpp_info.name = "OpenSSL"
+        self.cpp_info.components["crypto"].name = "Crypto"
+        self.cpp_info.components["crypto"].libs = ["libcrypto"]
+        self.cpp_info.components["ssl"].name = "SSL"
+        self.cpp_info.components["ssl"].libs = ["libssl"]
+
+You can also model system dependencies for each component or just header files. With this information, generators will be able to retrieve
+the information about each component and generate different targets for each of them (Still not supported by any generator).
+
+Examples of the usage of components in generators would be:
+
+- Generate different targets with appropriate names in CMake to iterate better with ``find_package()`` functionality. This will allow more
+  granularity when linking with packages with more than one library packaged.
+- Generate files with appropriate names for CMake config and package config files like: *xxx-config.cmake* or *xxx.pc*
```

Comment:
I'm not sure this is something we should mention. Probably all components should go in the same file (is it not possible for any generator?). If the components generate their own file, we can have collisions and Conan won't be able to handle them.

And, as this is totally speculative, I would vote to remove this _example_

Reply:
You are right. This was here when the feature was introduced together with the ``cpp_info.name`` attribute. But I think it is better to remove this from the reference right now and explain it better in a dedicated section in the future. Thanks!

Reply:
Yes, please remove that.